client.js

```diff
@@ -266,14 +276,17 @@ DHT.prototype.destroy = function (cb) {
 }
 
 /**
- * Add a DHT node to the routing table.
+ * Add a DHT node to the routing table unless the nodeID doesn't match
+ * their address and we're enforcing nodeID security.
```

Comment:
If you read the [enforcement section](http://bittorrent.org/beps/bep_0042.html#enforcement) of BEP42 you will notice that it only mentions lookups, not routing table maintenance.

I.e. as written it does not to exclude nodes from the overlay itself, only ignores them as store-targets to avoid eclipse attacks.